git.py

```diff
@@ -0,0 +1,162 @@
+import subprocess
+from functools import partial
+from typing import Optional, Iterable
+
+
+__all__ = [
+    'GitFuncs',
+    'GitFlows',
+]
+
+
+class BashFunc:
+    def __init__(self, func: Optional[str] = None, **kwargs):
+        self.func = func
+        self.kwargs = kwargs
+
+    def __call__(self) -> str:
+        if self.func is None:
+            return ''
+
+        print(f'> {self}')
+        output = subprocess.check_output(str(self), shell=True)
+        return output.decode('utf-8')
+
+    def __str__(self):
+        return self.func.format(**self.kwargs)
+
+
+class GitFuncs:
+    fetch = partial(
+        BashFunc,
+        'git fetch',
+    )
+    create_release_branch = partial(
+        BashFunc,
+        'git checkout -q -b {branch} --no-track origin/{source}',
+    )
+    delete_remote_branch = partial(
+        BashFunc,
+        'git push -q origin :{branch}',
+    )
+
+    cherry_pick = partial(
+        BashFunc,
+        'git cherry-pick -q {sha}',
+    )
+    commit = partial(
+        BashFunc,
+        'git commit -q --allow-empty -am "Release {version}"',
+    )
+    push = partial(
+        BashFunc,
+        'git push -q -u origin {branch}',
+    )
+    checkout = partial(
+        BashFunc,
+        'git checkout -q {branch}',
+    )
+    hard_reset = partial(
+        BashFunc,
+        'git reset -q --hard {branch}',
+    )
+
+    create_tag = partial(
+        BashFunc,
+        'git tag -q {version}',
+    )
+    push_tag = partial(
+        BashFunc,
+        'git push -q origin {version}',
```

Comment:
it has a different application:

> If you have a lot of tags that you want to push up at once, you can also use the --tags option to the git push command. This will transfer all of your tags to the remote server that are not already there.

here, I have only one tag that I need to push